### PR TITLE
Implement Ballerina Integrator VSCode plugin snippet generator

### PR DESCRIPTION
**Pull request by Nipunich**
## Purpose
Implement Ballerina Integrator VSCode plugin snippet generator by extending the Ballerina lang LSCompletion Provider SPI.

### CODE REVIEW

File: vscode-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Ballerina Integrator VSCode Plugin
+
+#### Adding new snippets to the plugin
+
+1. Create the relevant code snippet by following the format given in the file, SnippetGenerator.java.
```
Comment (dulanjalidilmi) [inline vscode-plugin/README.md]: Shall we provide the relative path of the `SnippetGenerator.java`?

File: vscode-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                            <tarLongFileMode>gnu</tarLongFileMode>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <properties>
+        <maven.checkstyleplugin.excludes>**/SourceGen.java</maven.checkstyleplugin.excludes>
```
Comment (Asitha) [inline vscode-plugin/pom.xml]: We need a parent pom?

File: vscode-plugin/src/main/java/org/wso2/integration/util/Snippets.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.wso2.integration.util;
+
+import org.wso2.integration.ballerinalangserver.SnippetsBlock;
+import org.wso2.integration.ballerinalangserver.SnippetsGenerator;
+
+public enum Snippets {
```
Comment (Asitha) [inline vscode-plugin/src/main/java/org/wso2/integration/util/Snippets.java]: JDocs?

File: vscode-plugin/README.md (hunk 1 above)
Comment (Nipunich) [reply vscode-plugin/README.md]: Fixed

File: vscode-plugin/src/main/java/org/wso2/integration/util/Snippets.java (hunk 3 above)
Comment (Nipunich) [reply vscode-plugin/src/main/java/org/wso2/integration/util/Snippets.java]: Fixed

File: vscode-plugin/pom.xml (hunk 2 above)
Comment (Nipunich) [reply vscode-plugin/pom.xml]: Fixed

**Pull request merged**